### PR TITLE
Use pod hostname as nodename

### PR DESCRIPTION
**Pull request by ruromero**

### CODE REVIEW

**Comment by pavolloffay** [comment]
@ruromero I need this changes in 5.x branch.

**Comment by ruromero** [comment]
@pavolloffay I'll do a backport

File: config/elasticsearch.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ cluster:
   name: elasticsearch
 network:
   host: 0.0.0.0
+node:
+  name: ${HOSTNAME}
```
Comment (pavolloffay) [inline config/elasticsearch.yml]: Is the hostname always equals to k8s `metadata.name`?

Comment (ruromero) [reply config/elasticsearch.yml]: yes https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods-hostname-and-subdomain-fields

**Pull request merged**